xscen/reduce.py

```diff
@@ -0,0 +1,144 @@
+import logging
+from typing import Union
+
+import numpy as np
+import xarray as xr
+import xclim.ensembles as xce
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["build_reduction_data", "reduce_ensemble"]
+
+
+def build_reduction_data(
+    datasets: Union[dict, list], *, xrfreqs: list = None, horizons: list = None
+) -> xr.DataArray:
+    """
+    Utility that constructs the input required for ensemble reduction. This will combine all variables into a single DataArray and
+    stack all dimensions except "realization".
+
+    Parameters
+    ----------
+    datasets: dict, list
+        Dictionary of datasets in the format {"id": dataset}, or list of datasets. This can be generated by calling .to_dataset_dict() on a catalog.
+    xrfreqs: list
+        List of unique frequencies across the datasets.
+        If None, the script will attempt to guess the frequencies from the datasets' metadata or with xr.infer_freq().
+    horizons: list
+        Subset of horizons on which to create the data.
+
+    Returns
+    -------
+    xr.DataArray
+        2D DataArray of dimensions "realization" and "criteria", to be used as input for ensemble reduction.
+    """
+
+    # Use metadata to identify the simulation attributes
+    info = {}
+    keys = datasets.keys() if isinstance(datasets, dict) else range(len(datasets))
+    for key in keys:
+        info[key] = {}
+        info[key]["id"] = datasets[key].attrs.get("cat:id", None) or key
+        info[key]["xrfreq"] = datasets[key].attrs.get("cat:xrfreq", None) or None
```

Comment:
```suggestion
        info[key]["xrfreq"] = datasets[key].attrs.get("cat:xrfreq") or xr.infer_freq(datasets[key].time)
```
Non? C'est ce que dit la docstring!